docs/marionette.collectionview.md

```diff
@@ -451,649 +512,587 @@ var CollectionView = Marionette.CollectionView({
-`View`s and `regions`. The following code will render the same table as in
-[Marionette 2](#tables-using-marionette-2):
+```js
+var MyChildView = Marionette.View.extend({
+  template: _.template('ChildView'),
+  onDestroy: function() {
+    console.log('I will get destroyed');
+  }
+})
 
-```javascript
-var RowView = Marionette.View.extend({
-  tagName: 'tr',
-  template: '#row-template'
+var myCollectionView = new Marionette.CollectionView({
+  childView: MyChildView,
+  collection: new Backbone.Collection([{ id: 1 }])
 });
 
-var TableBody = Marionette.CollectionView.extend({
-  tagName: 'tbody',
-  childView: RowView
-});
+myCollectionView.render();
 
-var TableView = Marionette.View.extend({
-  tagName: 'table',
-  className: 'table table-hover',
-  template: '#table',
+myCollectionView.destroy(); // logs "I will get destroyed"
+```
 
-  regions: {
-    body: {
-      el: 'tbody',
-      replaceElement: true
-    }
-  },
+## CollectionView's `filter`
 
-  onRender: function() {
-    this.showChildView('body', new TableBody({
-      collection: this.collection
-    }));
+`CollectionView` allows for a custom `filter` option if you want to prevent some of the
+underlying `collection`'s models from being rendered as child views.
+The filter function takes a model from the collection and returns a truthy value if the child should be rendered,
+and a falsey value if it should not.
+
+```js
+var cv = new Marionette.CollectionView({
+  childView: SomeChildView,
+  emptyView: SomeEmptyView,
+  collection: new Backbone.Collection([
+    { value: 1 },
+    { value: 2 },
+    { value: 3 },
+    { value: 4 }
+  ]),
+
+  // Only show views with even values
+  filter: function (child, index, collection) {
+    return child.get('value') % 2 === 0;
   }
 });
 
-var list = new Backbone.Collection([
-  {id: 1, text: 'My text'},
-  {id: 2, text: 'Another Item'}
-]);
+// renders the views with values '2' and '4'
+cv.render();
 
-var myTable = new TableView({
-  collection: list
+// change the filter
+// renders the views with values '1' and '3'
+cv.setFilter(function (child, index, collection) {
+  return child.get('value') % 2 !== 0;
 });
 
-myTable.render();
+// renders all views
+cv.removeFilter();
 ```
 
-We can leave the templates as-is for this example. The major advantage of this
-style is that we can create a region in any part of `TableView` as well as in
-`RowView` and treat it just as any independent widget.
-
-### Rendering Trees
-
-Tree structures are extremely useful layouts for nesting the same type of data
-over and over. A common example of this would be the Windows Explorer file
-picker.
+### CollectionView's `setFilter`
 
-#### Trees in Marionette 2
+The `setFilter` method modifies the `CollectionView`'s filter attribute, and
+renders the new `ChildViews` in a efficient way, instead of
+rendering the whole DOM structure again.
+A `preventRender` option will prevent the view to be rendered.
```

Comment:
This would read better as: Passing `preventRender: true` in the `options` argument will prevent the view being rendered.